src/agent/src/device.rs

```diff
@@ -379,6 +379,48 @@ pub async fn wait_for_pci_device(
     Ok(addr)
 }
 
+#[derive(Debug)]
+struct NetPciMatcher {
+    devpath: String,
+}
+
+impl NetPciMatcher {
+    fn new(relpath: &str) -> Result<NetPciMatcher> {
+        let root_bus = create_pci_root_bus_path();
+        Ok(NetPciMatcher {
+            devpath: format!("{}{}", root_bus, relpath),
+        })
+    }
+}
+
+impl UeventMatcher for NetPciMatcher {
+    fn is_match(&self, uev: &Uevent) -> bool {
+        uev.devpath == self.devpath && uev.subsystem == "net" && !uev.interface.is_empty()
+    }
+}
+
+pub async fn wait_for_net_interface(
```

Comment:
Could you add a few unit tests for this function? The final call to ` wait_for_uevent()` would probably be too awkward to test, but the remaining parts should be possible, I think?

Reply:
Done